url_shortener_proj/app/Http/Controllers/UrlController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Url;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Str;
+use App\Visitor;
+use Hash;
+use Session;
+
+
+class UrlController extends Controller
+{
+    public function store(Request $request)
+    {
+        $request->validate([
+            'original' => 'required|url'
+        ]);
+        $url = Url::create([
+            'original' => $request->original,
+            'short' => Str::random(7)
+        ]);
+        $short = $url->short;
+
+        Session::flash('success', 'The short URL was generated successfully!');
+
+        return redirect()->route('urls.details', $short);
+    }
+
+    public function details($short)
+    {
+        $urls = Url::all();
+        foreach ($urls as $u) {
+            if ($u->short == $short)
+                $url = $u;
+        }
+
+        if (!isset($url)) {
+            return view('errors.' . '500');
```

Comment:
Error with status code 500 means "Internal Server Error". In this case, it's more appropriate to return 404 error - Not found.

For complete reference, take a look at: https://www.restapitutorial.com/httpstatuscodes.html